app/src/main/java/org/dslul/openboard/inputmethod/latin/common/Colors.kt

```diff
@@ -24,39 +31,325 @@ import org.dslul.openboard.inputmethod.latin.KeyboardWrapperView
+        DrawableCompat.setTintMode(background, PorterDuff.Mode.MULTIPLY)
+        DrawableCompat.setTintList(background, colorStateList)
+    }
+
+    override fun getDrawable(type: BackgroundType, attr: TypedArray, context: Context): Drawable {
+        val drawable = when (type) {
+            BackgroundType.KEY, BackgroundType.ADJUSTED_BACKGROUND, BackgroundType.BACKGROUND,
+            BackgroundType.SUGGESTION, BackgroundType.ACTION_MORE_KEYS ->
+                attr.getDrawable(R.styleable.KeyboardView_keyBackground)
+            BackgroundType.FUNCTIONAL -> attr.getDrawable(R.styleable.KeyboardView_functionalKeyBackground)
+            BackgroundType.SPACE -> {
+                if (hasKeyBorders) attr.getDrawable(R.styleable.KeyboardView_spacebarBackground)
+                else attr.getDrawable(R.styleable.KeyboardView_spacebarNoBorderBackground)
+            }
+            BackgroundType.ACTION -> {
+                if (themeStyle == STYLE_HOLO && hasKeyBorders) // no borders has a very small pressed drawable otherwise
+                    attr.getDrawable(R.styleable.KeyboardView_functionalKeyBackground)
+                else
+                    attr.getDrawable(R.styleable.KeyboardView_keyBackground)
+            }
+        }?.mutate() ?: attr.getDrawable(R.styleable.KeyboardView_keyBackground)?.mutate()!! // keyBackground always exists
+
+        setBackgroundColor(drawable, type, context)
+        return drawable
+    }
+
+    override fun setKeyboardBackground(view: View, context: Context) {
+        when (view) {
+            is MoreSuggestionsView -> view.background.colorFilter = backgroundFilter
+            is MoreKeysKeyboardView ->
+                if (themeStyle != STYLE_HOLO)
+                    setBackgroundColor(view.background, BackgroundType.ADJUSTED_BACKGROUND, context)
+                else view.background.colorFilter = adjustedBackgroundFilter
+            is SuggestionStripView -> setBackgroundColor(view.background, BackgroundType.SUGGESTION, context)
+            is EmojiPageKeyboardView, // to make EmojiPalettesView background visible, which does not scroll
+            is MainKeyboardView -> view.setBackgroundColor(Color.TRANSPARENT) // otherwise causes issues with wrapper view when using one-handed mode
+            is KeyboardWrapperView, is EmojiPalettesView, is ClipboardHistoryView -> {
+                if (keyboardBackground != null) view.background = keyboardBackground
+                else view.background.colorFilter = backgroundFilter
+            }
+            else -> view.background.colorFilter = backgroundFilter
+        }
+    }
+}
+
+class DefaultColors (
+    override val themeStyle: String,
+    override val hasKeyBorders: Boolean,
+    private val accent: Int,
+    private val gesture: Int,
+    private val background: Int,
+    private val keyBackground: Int,
+    private val functionalKey: Int,
+    private val spaceBar: Int,
+    private val keyText: Int,
+    private val keyHintText: Int,
```

Comment:
`keyHintText` is marked as not used by Android Studio. What color is now used for the hint label?

Reply:
Oops; I went too fast with the ColorType assignment.
This has been corrected in the commit e6ade04.